### PR TITLE
feat: #10611 pkce

### PR DESCRIPTION
**Pull request by bashleigh**
# Pull request checklist

**Detail as per issue below (required):**

fixes: #10611

### CODE REVIEW

File: packages/connect-session/src/browser/index.ts [hunk 1]
```diff
@@ -222,16 +255,17 @@ export class ReapitConnectBrowserSession {
...
-  public connectAuthorizeRedirect(redirectUri?: string): void {
+  public async connectAuthorizeRedirect(redirectUri?: string): Promise<void> {
     const authRedirectUri = redirectUri || this.connectLoginRedirectPath
     const params = new URLSearchParams(window.location.search)
     params.delete('code')
     const search = params ? `?${params.toString()}` : ''
     const internalRedirectPath = encodeURIComponent(`${window.location.pathname}${search}`)
     const stateNonce = uuid()
     this.refreshTokenStorage.setItem(stateNonce, internalRedirectPath)
+    const code_challenge = await this.encryptCodeVerifier(this.codeVerifier)
```
Comment (willmcvay) [inline packages/connect-session/src/browser/index.ts:266]: Snake case `code_challenge` - it's not PHP :-)

File: packages/connect-session/src/browser/index.ts [hunk 2]
```diff
@@ -296,6 +330,8 @@ export class ReapitConnectBrowserSession {
               client_id: this.connectClientId,
               grant_type: 'authorization_code',
               code: this.authCode as string,
+              code_verifier: this.usePKCE ? this.codeVerifier : undefined,
```
Comment (joshbalfour) [inline packages/connect-session/src/browser/index.ts:333]:
```ts
new URLSearchParams({ a: undefined }).toString()
```
results in `"a=undefined"`, so does this still work if `usePKCE` is false?

File: packages/connect-session/README.md [hunk 3]
```diff
@@ -145,6 +145,21 @@ export const SomeComponent: React.FC = () => {
 }
 ```
 
+## PKCE 
+
+[Proof Key for Code Exchange](https://oauth.net/2/pkce/)
+
+By default PKCE is enabled. To disable PKCE, use `usePKCE` varaible on `ReapitConnectBrowserSession` and set it to false.
```
Comment (plittlewood-rpt) [inline packages/connect-session/README.md:152]: typo (varaible)

**Review by joshbalfour — APPROVED** [review]
LGTM 👍

**Pull request merged**